kratos/spatial_containers/bounding_volume_tree.h

```diff
@@ -270,11 +270,11 @@ class _6DOP : public kDOP // AABB
     KRATOS_CLASS_POINTER_DEFINITION(_6DOP);
     _6DOP() : kDOP() {this->Initialize();}
     virtual ~_6DOP() {}
-    virtual std::size_t NumberOfDirections() const {return 3;}
+    virtual std::size_t NumberOfDirections() const override {return 3;}
 
 private:
     static const double msDirection[][3];
-    virtual const double (*Direction() const)[3];
+    virtual const double (*Direction() const override)[3];
```

Comment:
We've been looking at this with @jcotela and the conclusion is that the line tries to return a variable[][3]. Since it seems to be returned by reference and for every implementation the size of the first array is different, it return its as a pointer. In other words it wants to return ```(*)[3]``` but this is syntactically incorrect.

Somehow the construction:
```C++
type (*Function() const)[x]
```

Works, but I personally suspects that this is an abuse of the notation and it was not taken into account while extending the parser for C++ decorators.

If one defines 

```C++
typedef const double BaseType[3];
typedef BaseType* ArrayType;
```

Then the code can be rewritten as:

```C++
# Header
virtual kDOP::ArrayType Direction() const override;

# Impl
kDOP::ArrayType _6DOP::Direction() const {return msDirection;}
```

Not 100% sure that it does the same but this way it compiles and can have the ```override``` decorator.

Anyone can try this out?